source/fundamentals/crud/write-operations/embedded-arrays.txt

```diff
@@ -140,81 +126,66 @@ in our update, we encounter the following error:
 
    The positional operator did not find the match needed from the query.
 
-.. warning::
-
-   Do not use the ``$`` operator in an ``upsert`` call because the ``$``
-   is treated as a field name in the insert document.
-
 .. _all-match-operator:
 
-Match All Array Elements
-------------------------
+Matching All Array Elements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 To perform the update on all of the array elements of each document that
-matches your query document in your update operation, use the all
-positional operator, ``$[]``.
+matches your query document, use the all positional ``$[]`` operator.
```

Comment:
issue: Two things. Throughout the page, we're saying "array elements of each document that matches your query document". I think we can safely drop the second use of 'document', e.g. "... document that matches your query".

Next, I think it's better to show the operator after giving it its name, so use the "all positional operator >operator<".